frontend/lib/src/components/core/Maybe/Maybe.test.tsx

```diff
@@ -14,85 +14,75 @@
  * limitations under the License.
  */
 
-import { ReactWrapper } from "enzyme"
 import React from "react"
-import { mount } from "@streamlit/lib/src/test_util"
+import { render } from "@streamlit/lib/src/test_util"
 import Maybe from "./Maybe"
 
 interface OuterProps {
   name: string
+  enable: boolean
 }
 
 interface InnerProps {
   name: string
 }
 const Inner = (props: InnerProps): any => <div>{props.name}</div>
 
+const Outer = (props: OuterProps): any => (
+  <Maybe enable={props.enable}>
+    <Inner name={props.name} />
+  </Maybe>
+)
+
 describe("The Maybe component", () => {
   describe("when enable is true", () => {
-    let component: ReactWrapper
-
-    beforeEach(() => {
-      const Outer = (props: OuterProps): any => (
-        <Maybe enable={true}>
-          <Inner name={props.name} />
-        </Maybe>
-      )
-      component = mount(<Outer name={"old again"} />)
-    })
-
     afterEach(() => {
-      component.unmount()
       jest.restoreAllMocks()
     })
 
     it("should invoke the render method when the props of an enclosing element update", () => {
+      const { rerender } = render(<Outer name={"old again"} enable={true} />)
+
       const spyShouldComponentUpdate = jest.spyOn(
         Maybe.prototype,
         "shouldComponentUpdate"
       )
       const spyRender = jest.spyOn(Maybe.prototype, "render")
-      component.setProps({ name: "new name" })
+
+      rerender(<Outer name={"new name"} enable={true} />)
+
       expect(spyShouldComponentUpdate).toHaveBeenCalled()
       expect(spyRender).toHaveBeenCalled()
     })
 
     it("should call render() when a Maybe is first disabled", () => {
+      const { rerender } = render(<Outer name={"old again"} enable={true} />)
+
       const spyShouldComponentUpdate = jest.spyOn(
         Maybe.prototype,
         "shouldComponentUpdate"
       )
       const spyRender = jest.spyOn(Maybe.prototype, "render")
-      component.setProps({ name: "new name", enable: false })
+
+      rerender(<Outer name={"new name"} enable={false} />)
```

Comment:
Updated here - on first render, `enable={true}` per the describe block, then checks that render is called when Maybe is first disabled, so rerender with `enable={false}`